airflow/providers/amazon/aws/hooks/elasticache.py

```diff
@@ -0,0 +1,265 @@
+        :return: Response from ElastiCache create replication group API
+        """
+        return self.conn.create_replication_group(**config)
+
+    def delete_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for deleting a replication group
+
+        :param replication_group_id: ID of replication group to delete
+        :return: Response from ElastiCache delete replication group API
+        """
+        return self.conn.delete_replication_group(ReplicationGroupId=replication_group_id)
+
+    def describe_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for describing a replication group
+
+        :param replication_group_id: ID of replication group to describe
+        :return: Response from ElastiCache describe replication group API
+        """
+        return self.conn.describe_replication_groups(ReplicationGroupId=replication_group_id)
+
+    def get_replication_group_status(self, replication_group_id):
+        """
+        Get current status of replication group
+
+        :param replication_group_id: ID of replication group to check for status
+        :return: Current status of replication group
+        """
+        return self.describe_replication_group(replication_group_id)[self.REPLICATION_GROUPS][0][self.STATUS]
+
+    def is_replication_group_available(self, replication_group_id):
+        """
+        Helper for checking if replication group is available or not
+
+        :param replication_group_id: ID of replication group to check for availability
+        :return: True if available else False
+        """
+        return self.get_replication_group_status(replication_group_id) == self.STATUS_AVAILABLE
+
+    def _has_reached_terminal_state(self, replication_group_id):
+        """
+        Helper for checking if we should stop poking replication group for availability or not
+
+        :param replication_group_id: ID of replication group to check
+        :return: Flag to check if availability check should be stopped or not and current status
+        """
+        status = self.get_replication_group_status(replication_group_id=replication_group_id)
+
+        return status in (
+            self.STATUS_AVAILABLE,
+            self.STATUS_CREATE_FAILED,
+            self.STATUS_DELETING
+        ), status
+
+    def wait_for_availability(
+        self,
+        replication_group_id,
+        initial_sleep_time=None,
+        exponential_back_off_factor=None,
+        max_retries=None
+    ):
+        """
+        Check if replication is available or not by performing a describe over it
+
+        :param max_retries: Max tries for checking availability of replication group
+        :param exponential_back_off_factor: Factor for deciding next sleep time
+        :param initial_sleep_time: Initial sleep time in seconds
+        :param replication_group_id: ID of replication group to check for availability
```

Comment:
Could you add types to these please?

```suggestion
        :param max_retries: Max tries for checking availability of replication group
        :type max_retries: int
        :param exponential_back_off_factor: Factor for deciding next sleep time
        :type exponential_back_off_factor: ...
        :param initial_sleep_time: Initial sleep time in seconds
        :type initial_sleep_time:
        :param replication_group_id: ID of replication group to check for availability
        :type replication_group_id:
```

Same for all the other public methods too please.